crates/crates_io_database/src/models/git_index_sync_queue_fetch_batch.sql

```diff
@@ -0,0 +1,18 @@
+WITH
+    batch AS (
+        SELECT
+            crate_name
+        FROM
+            git_index_sync_queue
+        ORDER BY
+            created_at ASC
+        FOR UPDATE
+        LIMIT
+            $1
+    )
+DELETE FROM git_index_sync_queue USING batch
```

Comment:
Just a note: 

using a postgres table as a queue like this has one problem: 

if our handler code fails, the entries are still removed from the queue. 
Also, multiple workers arent supported. 

If these things are an issue, you could do it the same way we do it in docs.rs: 

- start a transaction
- `SELECT [...] LIMIT 100 FOR UPDATE SKIP LOCKED` 
- do your work 
- `DELETE` the entries 
- commit the transaction

Through this, 
- other workers just skip over these locked records, and 
- and in case of an error, your transaction is rolled back, and the records are unlocked again.